pelican/plugins/more_categories/more_categories.py

```diff
@@ -75,9 +74,24 @@ def create_categories(generator):
 
     generator.categories = sorted(
         list(cat_dct.items()),
-        reverse=generator.settings.get('REVERSE_CATEGORY_ORDER') or False,
+        reverse=generator.settings.get("REVERSE_CATEGORY_ORDER") or False,
     )
-    generator._update_context(['categories'])
+    generator._update_context(["categories"])
+
+    # Add subcategories
+    cats = {}
```

Comment:
If you initialise this as `cats = defaultdict(list)`, you don't need to test whether it already contains a given category on L86.